lib/jekyll/site_data/file.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module Jekyll
+  module SiteData
+    class File
```

Comment:
In generally, I try to not use class names which match stdlib names, like `File`. It can just confuse things a bit. We could try `Jekyll::DataFile` and `Jekyll::DataDirectory`, or something which doesn't match the stdlib name `File`.